test/TestUtilities/Test.Utility/ProjectManagement/TestNuGetProjectContext.cs

```diff
@@ -17,17 +17,29 @@ public class TestNuGetProjectContext : IMSBuildNuGetProjectContext
     {
         private Guid _operationId;
         public TestExecutionContext TestExecutionContext { get; set; }
+        public Lazy<List<string>> Logs { get; } = new Lazy<List<string>>();
+        public bool EnableLogging { get; set; }
 
         public void Log(MessageLevel level, string message, params object[] args)
         {
             // Uncomment when you want to debug tests.
             // Console.WriteLine(message, args);
+
+            if (EnableLogging && args != null)
+            {
+                Logs.Value.Add(message + " " + string.Join(" ", args));
```

Comment:
Shouldn't you log the message even if args is null?

Reply:
On a second read, looks like you should be using String.Format(message, args).

Reply:
Refactored little bit for args is null case. But since I don't know actual format of message I can't use it. 
Also such fine details are not important for now.

Reply:
The format is well-defined at https://docs.microsoft.com/en-us/dotnet/api/system.string.format?view=netframework-4.7.2.

Reply:
The book "Writing High Performance .NET Code" explains how the Bing team got considerable perf improvements by doing this type of single-line string concatenation, rather than using `string.Format`. The problem with `string.Format` is it needs to parse the input string to find the `{x}` placeholders, which is work that `string.Concat` does not need to do.  So, in a general sense, I would not support replacing `str1 + " " + str2` with `string.format("{0} {1}", str1, str2)`.

Reply:
The method signature implies a format string with parameters. This leads me to believe that the desired behavior is to use String.Format.
Concatenation doesn't work for strings that are localized as different languages have different sentence structure and may need different ordering of placeholders.